### PR TITLE
作业

### PR DESCRIPTION
**Pull request by tianlinzhong**
更正

### CODE REVIEW

File: meter.py [hunk 1]
```diff
@@ -168,18 +168,41 @@ def on_put(req, resp, id_):
...
-
+        cursor.execute(" SELECT id  FROM tbl_meters WHERE id = id_")
+        if cursor.fetchone() is not None:
+            cursor.close()
+            cnx.disconnect()
+            raise falcon.HTTPError(falcon.HTTP_404, title='API.BAD_REQUEST',
+                                   description='API.METER_ID_IS_ALREADY_IN_USE')
         # TODO 4: check if meter name is in use by other meters
-
+        cursor.execute(" SELECT name FROM tbl_meters  WHERE id = id_")
```
Comment (guangyuzhang) [inline meter.py]: Wrong SQL statement.

File: meter.py [hunk 2]
```diff
@@ -168,18 +168,41 @@ def on_put(req, resp, id_):
...
+        else:
+            description = None
 
         cnx = mysql.connector.connect(**config.myems_demo_db)
         cursor = cnx.cursor()
 
         # TODO 3: check if meter id exists
-
+        cursor.execute(" SELECT id  FROM tbl_meters WHERE id = id_")
+        if cursor.fetchone() is not None:
```
Comment (guangyuzhang) [inline meter.py]: Wrong condition

File: meter.py [hunk 3]
```diff
@@ -168,18 +168,41 @@ def on_put(req, resp, id_):
...
+            cursor.close()
+            cnx.disconnect()
+            raise falcon.HTTPError(falcon.HTTP_404, title='API.BAD_REQUEST',
+                                   description='API.METER_ID_IS_ALREADY_IN_USE')
```
Comment (guangyuzhang) [inline meter.py]: Wrong description

**Comment by tianlinzhong** [comment]
#7

**Pull request closed without merging**